### PR TITLE
R: Fail CI/CD on NOTEs, check examples on UBSAN, log valgrind output

### PR DESCRIPTION
**Pull request by krlmlr**
to avoid discussions on CRAN. To finish, need to revise a7d95a32e. Please advise.

### CODE REVIEW

File: src/common/types/vector_constants.cpp [hunk 1]
```diff
@@ -9,8 +9,8 @@ namespace duckdb {
 // In the case of the FlatVector/ConstantVector, they point towards static regions of memory
 // Hence in this case these cause no problems, as the destructors are non-trivial but effectively nops
 #ifdef __clang__
-#pragma clang diagnostic push
-#pragma clang diagnostic ignored "-Wexit-time-destructors"
+//#pragma clang diagnostic push
+//#pragma clang diagnostic ignored "-Wexit-time-destructors"
```
Comment (Mytherin) [inline src/common/types/vector_constants.cpp:13]: I think these are required in the OSX builds currently, see [here](https://github.com/duckdb/duckdb/blob/master/src/CMakeLists.txt#L12).

Perhaps we should just remove the warning entirely, or explicitly disable the warning for these files using the CMake instead.

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! Looks good to me. One comment:

File: src/common/types/vector_constants.cpp (hunk 1 above)
Comment (krlmlr) [reply src/common/types/vector_constants.cpp:13]: Thanks. One file is ours, the other is vendored. What's the process of updating the vendored version (if we wanted to erase the check there)?

Comment (Mytherin) [reply src/common/types/vector_constants.cpp:13]: We added the check there, so update in any way you want

**Comment by krlmlr** [comment]
I'll take a look soon.

Following @hannes's question regarding non-detection of Valgrind errors: we do run the Valgrind checks on CI/CD, but we need also to parse the logs: https://github.com/krlmlr/duckdb/runs/6776149490?check_suite_focus=true#step:8:3046.

I'll take a look at the UBSAN checks too.

**Comment by krlmlr** [comment]
This now fails the UBSAN check. We don't parse valgrind output yet, but it's logged.

**Comment by Mytherin** [comment]
#3812 should fix the valgrind issues that are there but have not been uncovered before

**Comment by krlmlr** [comment]
As expected: the UBSAN build now fails, consistently with CRAN.

**Comment by krlmlr** [comment]
Valgrind log: https://github.com/duckdb/duckdb/runs/6823490797?check_suite_focus=true#step:13:105

**Comment by krlmlr** [comment]
Pulled in the changes from https://github.com/Mytherin/duckdb/tree/krlmlr-f-r-debug. Builds were green there.

**Review by hannes — APPROVED** [review]

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**